tools/fastp/fastp.xml

```diff
@@ -0,0 +1,332 @@
+        </section>
+        <section name="base_correction_options" title="Base correction by overlap analysis options" expanded="False">
+            <param name="correction" type="boolean" truevalue="-c" falsevalue="" checked="false" label="Enable base correction" argument="-c" help="Enable base correction in overlapped regions (only for PE data), default is disabled."/>
+        </section>
+        <section name="umi_processing" title="UMI processing" expanded="False">
+            <param name="umi" type="boolean" truevalue="-U" falsevalue="" checked="false" label="Enable unique molecular identifer" argument="-U" help="Enable unique molecular identifer (UMI) preprocessing."/>
+            <param name="umi_loc" type="text" optional="true" label="UMI location" argument="--umi_loc" help="Specify the location of UMI, can be (index1/index2/read1/read2/per_index/per_read, default is none."/>
+            <param name="umi_len" type="integer" optional="true" label="UMI length" argument="--umi_len" help="If the UMI is in read1/read2, its length should be provided."/>
+            <param name="umi_prefix" type="text" optional="true" label="UMI prefix" argument="--umi_prefix" help="If specified, an underline will be used to connect prefix and UMI (i.e. prefix=UMI, UMI=AATTCG, final=UMI_AATTCG). No prefix by default."/>
+        </section>
+        <section name="overrepresented_sequence_analysis" title="Overrepresented sequence analysis" expanded="False">
+            <param name="overrepresentation_analysis" type="boolean" truevalue="-p" falsevalue="" checked="false" label="Enable overrepresented analysis" argument="-p" help="Enable overrepresented sequence analysis."/>
+            <param name="overrepresentation_sampling" type="integer" optional="true" label="Overrepresentation sampling" argument="-P" help="One in (--overrepresentation_sampling) reads will be computed for overrepresentation analysis (1~10000), smaller is slower, default is 20."/>
+        </section>
+    </inputs>
+
+    <outputs>
+        <data format="fastq" name="out1" label="${tool.name} output"/>
+        <data format="fastq" name="out2" label="${tool.name} paired end output">
+            <filter>single_paired['single_paired_selector'] == "paired"</filter>
+        </data>
+    </outputs>
+
+    <tests>
+        <test>
+            <param name="in1" value="R1.fq"/>
+            <param name="single_paired_selector" value="single"/>
+            <output name="out1" file="out1.fq" ftype="fastq"/>
+        </test>
+        <test>
+            <param name="in1" value="R1.fq"/>
+            <param name="single_paired_selector" value="single"/>
+            <param name="adapter_sequence1" value="ATCG"/>
+            <output name="out1" file="out_a.fq" ftype="fastq"/>
+        </test>
+        <test>
+            <param name="in1" value="R1.fq"/>
+            <param name="single_paired_selector" value="single"/>
+            <section name="umi_processing">
```

Comment:
I don't think you need to state the section within the test, it should be enough to give the param name.

Reply:
But its good to state them, better explicit then implicit :)